comp/dogstatsd/listeners/udp.go

```diff
@@ -18,22 +17,11 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-var (
-	udpExpvars             = expvar.NewMap("dogstatsd-udp")
-	udpPacketReadingErrors = expvar.Int{}
-	udpPackets             = expvar.Int{}
-	udpBytes               = expvar.Int{}
-)
+var udpTelemetry = newListenerTelemetry("udp", "UDP")
```

Comment:
This is still global state with a named `init` function that still runs before main. How is this better?

Reply:
Thanks, @vickenty you are right. 

I'm getting familiar with Go, `init()`, and global state. I got confused and thought that `init()` functions where different than named functions. In turn, that is not 😄. I updated the PR title and description. 

As to your question, `How is this better?`

Now that functionality, rather than being spread out into `init()` and accessing multiple global variables, we have a single global variable per listener that handles everything related to telemetry.

Reply:
> Now that functionality, rather than being spread out into init() and accessing multiple global variables, we have a single global variable per listener that handles everything related to telemetry.

This is a summary of the changes you did. What I'd like to know is what was the problem that this PR solves? Having global data may be a problem, but the PR doesn't address that. What else?

Reply:
> What I'd like to know is what problem this PR solves?

We DRY our code base by using the existing functionality from `newListenerTelemetry`. 

The PR doesn't solve the problem of global state but refactors the codebase. It ensures that each listener declares their global state in a consistent way.